clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -440,6 +440,7 @@ private void handleCompletedReceives(List<ClientResponse> responses, long now) {
             correlate(req.request().header(), header);
             if (!metadataUpdater.maybeHandleCompletedReceive(req, now, body))
                 responses.add(new ClientResponse(req, now, false, body));
+            inFlightRequests.completeNext(source);
```

Comment:
@zhuchen1018 : But the issue is that we already processed the response. Next time, when we come back to handleCompletedReceives() again, we will get a response intended for a request different from what's in the head of inFlightRequests.